vm/src/symboltable.rs

```diff
@@ -0,0 +1,272 @@
+/* Python code is pre-scanned for symbols in the ast.
+
+This ensures that global and nonlocal keywords are picked up.
+Then the compiler can use the symbol table to generate proper
+load and store instructions for names.
+*/
+
+use rustpython_parser::ast;
+use std::collections::HashMap;
+
+pub enum SymbolRole {
+    Global,
+    Nonlocal,
+    Used,
+    Assigned,
+}
+
+pub struct SymbolTable {
+    // TODO: split-up into nested scopes.
```

Comment:
Aha, I was wondering why there was only a single symbol table.

Reply:
There is a nested tree of symbol tables. Each symbol table can have multiple sub scopes, which is done now.